Dev multitapered spectral analysis

mer0mingian opened the pull request:
Import multitapered spectral analysis with accelerated cython computation.

alperyeg commented:
@mer0mingian please merge the `setup_adapted.py` to `setup.py`.

alperyeg commented on elephant/test/test_multitaper_utils.py:
```diff
@@ -0,0 +1,196 @@
+from numpy.testing import assert_almost_equal
+import numpy.linalg as la
+
+# TODO: function to test the jackknife procedure
```
What about this TODO?

mer0mingian replied:
Comment was deprecated (test function following directly on line 50sth)

alperyeg commented on elephant/_cython_utils.pyx:
```diff
@@ -0,0 +1,64 @@
+        t = ew[k - 1]
+        ew[k - 1] = t / dw[k - 1]
+        dw[k] = dw[k] - t * ew[k - 1]
+    for k in xrange(1, N):
```
`xrange` will fail for Python3. If the range of the list is not to big `range` should be sufficient.  Otherwise you could use the range from the six module.

mer0mingian replied:
fixed

alperyeg commented on elephant/test/test_multitaper.py:
```diff
@@ -0,0 +1,397 @@
+    n_freqs : size of frequency grid
+    sides : {'onesided', 'twosided'}
+       compute frequencies between [-PI,PI), or from [0, PI]
+    Returns
```
Can you add spaces here 
```python
"""
     compute frequencies between [-PI,PI), or from [0, PI]

Returns
-------
"""
```

mer0mingian replied:
fixed.

alperyeg commented:
@mer0mingian any update on this PR?

mer0mingian commented:
Sorry, heavy workload. All suggestions/requests treated.

alperyeg commented:
@mer0mingian thanks for the updates. But there is still one test failing, can you check please? Thanks! Afterwards I will go through again.

mer0mingian commented:
Fixed the last test failing and included the Matlab test data in the 'setup.py' file. Removed 'setup-adapted.py'.
Does it signify something that Travis CI still failes?

alperyeg commented on setup.py:
```diff
@@ -21,7 +21,9 @@
                                   os.path.join('current_source_density_src', 'LICENSE'),
                                   os.path.join('current_source_density_src', 'README.md'),
-                                  os.path.join('current_source_density_src', '*.py')]},
+                                  os.path.join('current_source_density_src', '*.py')
```
A comma is missing here.

dizcza commented:
Obsolete.
Not sure what did you come to 2 years ago, but adding cython files should have a reason. To clarify, speed-up is not certainly a reason (because it comes with the burden of nonunderstandable, hard to maintain code).
Even though we do need multitapers, there are pythonic "eco-free" implementations as well.
We look forward to have it in python.
Thank you.
